### DISCUSSION TITLE
How to change just the cloud transparency

### DISCUSSION THREAD

**Discussion by Illegit** [freeplane/freeplane#1894] [Help] [answered]
If i need to change only the cloud transparency of a group of selected nodes (and if there isn't any cloud it must  add 1 std cloud with the required transparency), which command I have to use? I tried with Copilot but I do not think it is able to utilize the Freeplane API... no result
Any suggestion?  Thanks

**Comment by Illegit** [comment]
Here one of the copilot solution... no hope: import org.freeplane.features.cloud.CloudModel
import org.freeplane.features.cloud.mindmapmode.MCloudController
import org.freeplane.features.map.NodeModel
import org.freeplane.features.mode.Controller

def void setCloudWithAlpha(NodeModel node, float alpha) {
    MCloudController cloudController = (MCloudController) Controller.getCurrentModeController().getExtension(CloudModel.class)
    CloudModel cloud = cloudController.createCloud(node.getMap())
    cloud.setAlpha(alpha)
    cloudController.setCloud(node, cloud)
}

Controller.getCurrentModeController().getMapController().getSelectedNodes().each { node ->
    setCloudWithAlpha(node, 0.75)
}

**Comment by Illegit** [comment]
Here another (not working):
import org.freeplane.features.cloud.CloudModel
import org.freeplane.features.cloud.mindmapmode.MCloudController
import org.freeplane.features.map.NodeModel
import org.freeplane.features.mode.Controller

def void setCloudWithAlpha(NodeModel node, float alpha) {
    MCloudController cloudController = (MCloudController) Controller.getCurrentModeController().getExtension(CloudModel.class)
    CloudModel cloud = cloudController.createCloud(node.getMap())
    cloud.setAlpha(alpha)
    cloudController.setCloud(node, cloud)
}

Controller.getCurrentModeController().getMapController().getSelectedNodes().each { node ->
    setCloudWithAlpha(node, 75/255.0)
}

**Comment by macmarrum** [comment, marked as answer]
I would go with the following script

```groovy
import java.awt.Color

// target alpha (0 - 255)
int ALPHA = 16

c.selecteds.each { n ->
    def cloudColor = n.cloud.color
    
    if (!cloudColor) {
        // set the default cloud color from Freeplane settings
        n.cloud.enabled = true
        // get its color
        cloudColor = n.cloud.color
    }

    if (cloudColor.alpha != ALPHA) {
        n.cloud.color = new Color(cloudColor.red, cloudColor.green, cloudColor.blue, ALPHA)
    }
}
```

**Reply by Illegit** [reply]
Thanks macmarrum, I learned something new and simple. Very appreciated. It works exactly as expected.
